sdk/core/azure_core/src/cloud.rs

```diff
@@ -0,0 +1,118 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+//! Azure cloud configuration.
+
+use std::{any::TypeId, collections::HashMap};
+
+/// Configurations for different Azure clouds.
+#[derive(Clone, Debug, Default, Eq, PartialEq)]
+#[non_exhaustive]
+pub enum CloudConfiguration {
+    /// Azure Public Cloud
+    #[default]
+    AzurePublic,
+
+    /// Azure Government
+    AzureGovernment,
+
+    /// Azure in China
+    AzureChina,
+
+    /// A custom cloud.
+    ///
+    /// # Example
+    ///
+    /// ```
+    /// # mod azure_service_module {
+    /// #   pub struct Audience;
+    /// # }
+    ///
+    /// use azure_core::{
+    ///     cloud::{Audiences, CloudConfiguration, CustomConfiguration},
+    ///     http::ClientOptions,
+    /// };
+    ///
+    /// let mut custom = CustomConfiguration::default();
+    /// custom.audiences = Audiences::new().with::<azure_service_module::Audience>("https://service.mycloud.local".to_string());
```

Comment:
No doubt this would reformat, which is nice for docs with limited width in tooltips, for example:

```suggestion
    /// custom.audiences = Audiences::new()
    ///     .with::<azure_service_module::Audience>("https://service.mycloud.local".to_string());
```